src/ApiCore/GapicClientTrait.php

```diff
@@ -93,6 +93,45 @@ private static function getGapicVersion(array $options)
         return self::$gapicVersion;
     }
 
+    /**
+     * Get default options. This function should be "overridden" by clients using late static
+     * binding to provide default options to the client.
+     *
+     * @return array
+     */
+    private static function getClientDefaults()
```

Comment:
I'm debating whether we want to keep this static or not. If we give up the fact it is static we can define it as abstract and have this be an enforceable contract. On the other hand, the only code using this is generated and will always end up implementing the method, so the contract isn't that useful. WDYT?

Reply:
I don't have a strong opinion. In other circumstances I would favour making this not static - but it is used in the client constructor, and calling non-static methods from the constructor is also not ideal.

Reply:
Okay, let's leave as is for now then :).